versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/CommitLogEntry.java

```diff
@@ -89,4 +89,8 @@ static CommitLogEntry of(
         .addAllKeyListsIds(keyListIds)
         .build();
   }
+
+  static CommitLogEntry withNewMeta(CommitLogEntry sourceCommit, ByteString metadata) {
```

Comment:
Can simplify by just adding a `CommitLogEntry withMetadata(ByteString metadata)` to the `CommitLogEntry` interface. (see e.g. `DatabaseAdapterConfig` interface)